File: app/src/main/java/im/dacer/kata/util/engine/SegmentEngine.java
/*
 * The MIT License (MIT)
 * Copyright (c) 2016 baoyongzhang <baoyz94@gmail.com>
 */
package im.dacer.kata.util.engine;

import im.dacer.kata.util.segment.BigBang;
import im.dacer.kata.util.segment.Parser;
import im.dacer.kata.util.segment.parser.KuromojiParser;

/**
 * Created by baoyongzhang on 2016/10/28.
 */
public class SegmentEngine {

    private static Parser getSegmentParser() {
        return new KuromojiParser();
    }

    public static void setup() {
        if (BigBang.initialized()) return;
        Parser parser = SegmentEngine.getSegmentParser();
        BigBang.setSegmentParser(parser);
    }
}

File: app/src/main/java/im/dacer/kata/util/engine/SegmentEngine.kt
package im.dacer.kata.util.engine

import android.content.Context
import im.dacer.kata.data.local.MultiprocessPref
import im.dacer.kata.data.model.segment.KanjiResult
import im.dacer.kata.util.segment.BigBang
import im.dacer.kata.util.segment.Parser

object SegmentEngine {

    private fun getSegmentParser(context: Context): Parser<List<KanjiResult>> {
        val pref = MultiprocessPref(context)
        return pref.segmentParser
    }

    fun setup(context: Context, force: Boolean = false) {
        if (BigBang.initialized() && !force) return
        val parser = getSegmentParser(context)
        BigBang.setSegmentParser(parser)
    }
}
